lib/rock/packaging/packageinfoask.rb

```diff
@@ -0,0 +1,177 @@
+
+module Autoproj
+    module Packaging
+        class PackageInfoAsk
+
+            class << self
+                alias :class_new :new
+
+                def new(which, options)
```

Comment:
PackageInfoAsk is an abstract base class to the package info providers, and thus mostly there for documentation purposes. The new() method has no real purpose in abstract classes, and, together with the initialize method, would generally raise an exception, if not for the factory semantics new() has now. If that new() override is problematic, i'd rather have a differently named class method do the instantiation and have initialize raise an exception than doing basically \@delegate.method() for every method, or going the method_missing route(which would require to remove all method defs).